app/frontend/js/modals/transformationDefinitionSettingsModal.js

```diff
@@ -57,38 +37,46 @@ if (transformationDefinitionSettingsForms) {
       let format = transformationDefinitionPreviewData.dataset.format;
       let completed = transformationDefinitionPreviewData.dataset.completed;
 
-      if (format == "JSON") {
-        result = JSON.stringify(JSON.parse(result), null, 2);
-      } else if (format == "XML") {
-        result = xmlFormat(result, {
-          indentation: "  ",
-          lineSeparator: "\n",
-        });
-      }
-
-      editor(`#js-record-selector-result-${id}`, format, true, result);
-      tooltip.disable();
+      if (result == "") {
```

Comment:
There are a lot of nested if else, and this file is pretty large and complicated. Could it be broken up to make it more readable?

Reply:
I'm guessing some of it could be shared functionality like those other simpler modals?

Reply:
Iv split this one up :)